app/src/main/java/com/x8bit/bitwarden/MainViewModel.kt

```diff
@@ -149,8 +154,11 @@ class MainViewModel @Inject constructor(
         sendEvent(MainEvent.CompleteAutofill(cipherView = action.cipherView))
     }
 
-    private fun handleCurrentUserStateChange() {
+    private fun handleCurrentUserStateChange(action: MainAction.Internal.CurrentUserStateChange) {
         recreateUiAndGarbageCollect()
+        if (action.isLoggedIn) {
+            specialCircumstanceManager.clearSpecialCircumstanceAfterLogin()
```

Comment:
Would it make more sense for the `SpecailCircumstanceManager to handle this itself:

```kotlin
class SpecialCircumstanceManagerImpl 
   ...
   
    init {
        authDiskSource
            .userAccountTokensFlow
            .filter { userAccountTokensList ->
                userAccountTokensList.any {
                    it.userId == authDiskSource.userState?.activeUserId && it.isLoggedIn
                }
            }
            .onEach {
                if (blah) {
                    specialCircumstance = null
                }
            }
            .launchIn()
    }
```

Reply:
I really like this!